src/runtime_src/xocl/core/memory.h

```diff
@@ -488,7 +488,7 @@ class buffer : public memory
       // allocate sufficiently aligned memory and reassign m_host_ptr
       if (posix_memalign(&m_host_ptr,alignment,sz))
         throw error(CL_MEM_OBJECT_ALLOCATION_FAILURE);
-    if (flags & CL_MEM_COPY_HOST_PTR)
+    if (flags & CL_MEM_COPY_HOST_PTR && host_ptr)
```

Comment:
I'm not sure about this fix. Why check host_ptr == NULL here? It does not make sense for user to set CL_MEM_COPY_HOST_PTR and pass in NULL host_ptr, right?

Reply:
It doesn't, and I think the callers do the right thing, but GCC 8.3 isn't smart enough to realise this and so bombs out with "-Werror" due to not knowing host_ptr is definitely null in this case. Adding the explicit check seemed to be the simplest way to fix the build without losing the check generally.

Reply:
Before the fix, if host_ptr is null, the program will crash, which is good. After the fix, we'll just silently skip copying? That does not sound right?